archinstall/lib/menu/selection_menu.py

```diff
@@ -220,150 +178,65 @@ def enable(self, selector_name, omit_if_set=False):
-
-		missing = 0
-		if not check('bootloader'):
-			missing += 1
-		if not check('hostname'):
-			missing += 1
-		if not check('audio'):
-			missing += 1
-		if not check('timezone'):
-			missing += 1
-		if not check('!root-password') and not check('!superusers'):
-			missing += 1
-		if not check('harddrives'):
-			missing += 1
-		if check('harddrives'):
-			if not self._menu_options.get('harddrives').is_empty() and not check('disk_layouts'):
-				missing += 1
-
-		return missing
-
-	def _set_root_password(self):
-		prompt = 'Enter root password (leave blank to disable root & create superuser): '
-		password = archinstall.get_password(prompt=prompt)
-
-		if password is not None:
-			self._menu_options.get('!superusers').set_current_selection(None)
-			archinstall.arguments['!users'] = {}
-			archinstall.arguments['!superusers'] = {}
-
-		return password
-
-	def _select_harddrives(self):
-		old_haddrives = archinstall.arguments.get('harddrives')
-		harddrives = archinstall.select_harddrives()
-
-		# in case the harddrives got changed we have to reset the disk layout as well
-		if old_haddrives != harddrives:
-			self._menu_options.get('disk_layouts').set_current_selection(None)
-			archinstall.arguments['disk_layouts'] = {}
-
-		if not harddrives:
-			prompt = 'You decided to skip harddrive selection\n'
-			prompt += f"and will use whatever drive-setup is mounted at {archinstall.storage['MOUNT_POINT']} (experimental)\n"
-			prompt += "WARNING: Archinstall won't check the suitability of this setup\n"
-
-			prompt += 'Do you wish to continue?'
-			choice = Menu(prompt, ['yes', 'no'], default_option='yes').run()
-
-			if choice == 'no':
-				return self._select_harddrives()
-
-		return harddrives
-
-	def _secret(self, x):
-		return '*' * len(x)
-
-	def _select_profile(self):
-		profile = archinstall.select_profile()
-
-		# Check the potentially selected profiles preparations to get early checks if some additional questions are needed.
-		if profile and profile.has_prep_function():
-			namespace = f'{profile.namespace}.py'
-			with profile.load_instructions(namespace=namespace) as imported:
-				if not imported._prep_function():
-					archinstall.log(' * Profile\'s preparation requirements was not fulfilled.', fg='red')
-					exit(1)
-
-		return profile
-
-	def _create_superuser_account(self):
-		superuser = archinstall.ask_for_superuser_account('Create a required super-user with sudo privileges: ', forced=True)
-		return superuser
+		# we allow for an callback to make something before the selector function is invoked
+		if self.pre_process_callback:
+			self.pre_process_calback(self,selector_name)
+		result = None
+		if selector.func:
+			result = selector.func()
+			self._menu_options[selector_name].set_current_selection(result)
+			self._data_store[selector_name] = result
+		# we allow for a callback after we get the result
+		if self.post_process_callback:
+			self.post_process_callback(self,selector_name,result if result else None)
```

Comment:
> You can pass just the result without having to do a `if ... else` since it's set to `None` already
I was the one which put the initialization of result, and had to do, because this conditional ... Changed